Zhengxinqi/plugins refractory

blizzardzheng opened the pull request:
重构插件机制

blizzardzheng committed:
重构 plugins 核心

blizzardzheng committed:
重构initialise state

vangie commented on app/backendAPI/packageAPI.js:
```diff
@@ -1,10 +1,11 @@
-import { request } from '../utils'
 import axios from 'axios'
+import { request } from '../utils'
+import { PluginsCache } from 'utils/plugins'
```
这个叫 Cache 不太准确，叫 Registry 更合适些。

blizzardzheng committed:
1.重命名 pluginRegistry
2.去除所有仅平台版使用到的接口，放到接口项目里
3.position 常量该用全大写的方式
4.otheraction 改名 callback
5.恢复误删的一行 context menu container
6.重命名loadInnerPlugin
7.重构 initialise step cache 的遍历方式，增加insert，move，multiple delete方法，
8.去除step Cache中无用的key字段

blizzardzheng committed:
去除无用代码
